tcmur_cmd_handler.c

```diff
@@ -2269,6 +2373,15 @@ static int tcmur_cmd_handler(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 		break;
 	}
 
+	rcmd_state = calloc(1, sizeof(*rcmd_state));
+	if (!rcmd_state) {
+		tcmu_dev_err(dev, "Failed to calloc memory for tcmur_cmd_state!\n");
+ 		ret = TCMU_STS_NO_RESOURCE;
+		goto untrack;
+	}
+
+	tcmu_cmd_set_private(cmd, rcmd_state);
+
 	switch(cdb[0]) {
```

Comment:
If we fail early in one of the handle_* functions where the cmd is never sent to the handler then we will leak the rcmd_state allocated above.

At the end of this function in the chunk:

```
        if (ret != TCMU_STS_ASYNC_HANDLED) {
                track_aio_request_finish(rdev, NULL);
                free(rcmd_state);
```

we also need to free rcmd_state.